btx/processing/rawimagetimetool.py

```diff
@@ -0,0 +1,575 @@
+        # self.det = psana.Detector(self.get_detector_name(run), self.ds.env())
+        self.det = psana.Detector(self.get_tt_name(), self.ds.env())
+
+    def get_tt_name(self) -> str:
+        """! Run psana.MPIDataSource.detnames() to get associated detectors,
+        process the output and find the name of the timetool detector.
+
+        @return detname (str) Name of the timetool detector (full or alias).
+        """
+        for tup in self.ds.detnames():
+            for name in tup:
+                if any(alias in name.lower() for alias in ['opal', 'timetool']):
+                    return name
+
+    def get_detector_name(self, run:str) -> str:
+        """! This method is deprecated.
+        Run detnames from shell and determine the time tool detector name.
+
+        @param run (str) Experiment run to check detectors for
+        @return detname (str) Name of the detector (full or alias)
+        """
+        # Define command to run
+        prog = 'detnames'
+        arg = f'exp={self.expmt}:run={run}'
+        cmd = [prog, arg]
+
+        # Run detnames, save output
+        output = subprocess.check_output(cmd)
+
+        # Convert to list and strip some characters
+        output = str(output).split()
+        output = [o.strip('\\n') for o in output]
+
+        # Select detectors related to timetool (Opal, e.g.)
+        # Need to see if this actually works for all hutches, seems to be good
+        # for cxi and mfx
+        ttdets = [det for det in output if "opal" in det.lower() or "timetool" in det.lower()]
+
+        # Return the first entry (could use any)
+        return ttdets[0]
+
+
+    def calibrate(self, run: str, order: int = 2, figs: bool = True):
+        """! Fit a calibration model to convert delay to time tool jitter
+        correction. MUST be run before analysis if a model has not been previously
+        fit. The fitted polynomial model is saved to a text file for later use.
+
+        @param run (str) SINGLE run number for the experimental calibration run.
+        @param order (int) Order of the polynomial to fit for calibration.
+        @param figs (bool) Whether or not to print diagnostic figures.
+        """
+
+        # Open calibration run first!
+        self.open_run(run)
+
+        # Detect the edges
+        delays, edges, ampls = self.process_run()
+        self.fit_calib(delays, edges, ampls, None, order)
+
+        # Write necessary files
+        run = self.format_run()
+        outdir = f'{self.savedir}/calib'
+        ## Model coefficients
+        fname = f'{run}.out'
+        self.write_file(self._model, fname, outdir)
+        ## Fitted edges
+        fname = f'EdgesFit_{run}.out'
+        self.write_file(self.edges_fit, fname, outdir)
+        ## Fitted delays
+        fname = f'DelaysFit_{run}.out'
+        self.write_file(self.delays_fit, fname, outdir)
+
+        if figs:
+            self.plot_calib(delays, edges, self._model)
+            self.plot_hist(self.edges_fit)
```

Comment:
Really nice! We should probably be more descriptive in the docstring about what's being done or at least point to the read-the-docs page once we have it.